src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminal.accessibility.contribution.ts

```diff
@@ -129,18 +130,27 @@ export class TerminalAccessibleViewContribution extends Disposable implements IT
 			return;
 		}
 		if (!this._configurationService.getValue(TerminalSettingId.AccessibleViewFocusOnCommandExecution)) {
-			this._onCommandExecutedShowListener.clear();
+			this._onDidRunCommand.clear();
+			this._onDidPreCommandFinishWindows.clear();
 			return;
-		} else if (this._onCommandExecutedShowListener.value) {
+		} else if (this._onDidRunCommand.value) {
 			return;
 		}
 
 		const capability = this._instance.capabilities.get(TerminalCapability.CommandDetection)!;
-		this._onCommandExecutedShowListener.value = this._register(capability.onCommandExecuted(() => {
+		this._onDidRunCommand.value = this._register(capability.onCommandExecuted(() => {
```

Comment:
How about this instead?

```ts
Event.any(capability.onCommandExecuted, capability.onCommandFinished)(() => {
  // shared listener
);
```

Reply:
Sure, they're mutually exclusive I think, but sounds good